src/events/SDL_events.c

```diff
@@ -796,6 +876,24 @@ SDL_WaitEventTimeout(SDL_Event * event, int timeout)
     }
 }
 
+static int
+SDL_SendWakeupEvent()
+{
+    SDL_VideoDevice *_this = SDL_GetVideoDevice();
+    if (!_this || !_this->SendWakeupEvent) {
+        return 0;
+    }
+    if (!_this->wakeup_lock || SDL_LockMutex(_this->wakeup_lock) == 0) {
```

Comment:
I hate to have to acquire this lock for every event just to see that nobody is waiting (in the common case).

Maybe you could use `SDL_AtomicGetPtr(&_this->wakeup_window)` (and `SDL_AtomicSetPtr()` when setting wakeup_window) to short-circuit this logic if nobody is waiting?

Reply:
Ok, I guess the atomic function are better but I am not implementing this change right away because I need more time to do it carefully so I leave that for later if you don't mind.